activerecord/test/cases/associations_test.rb

```diff
@@ -521,6 +524,21 @@ def test_preload_grouped_queries_of_through_records
     end
   end
 
+  def test_preload_through_records_with_already_loaded_middle_record
+    member = members(:groucho)
+    expected_member_detail_ids = member.organization_member_details_2.pluck(:id)
+
+    member.reload.organization # load through record
+
+    assert_queries(1) do
+      ActiveRecord::Associations::Preloader.new(records: [member], associations: :organization_member_details_2).call
```

Comment:
Down from 3 queries to 1 🎉
<img width="1158" alt="Screenshot 2022-11-30 at 6 28 56 pm" src="https://user-images.githubusercontent.com/54629302/204804040-bd988501-3790-4d4e-a8eb-d28f2938a73d.png">